include/DcmTrajectoryGenerator.h

```diff
@@ -0,0 +1,391 @@
+ */
+class SingleSupportTrajectory : public GeneralSupportTrajectory
+{
+ private:
+  
+  iDynTree::Vector2 m_dcmIos; /**< Desired position of the DCM at the beginning of the step */
+  iDynTree::Vector2 m_zmp; /**< Desired position of the ZMP at the beginning of the step */
+  double m_omega; /**< Time constant of the 3D-LIPM */
+
+  /** 
+   * Absolute "init" time of the Single Support phase
+   * it is important to notice that t0 is equal to the absolute time of the previous half
+   * DS phase. 
+   * Note that m_t0 is NOT equal to the start_time (aka the time of the beginning
+   * of the SS trajectory. Because the SS trajectory is obtained under the hypotesis of
+   * no DS phase).
+   */
+  double m_t0;
+
+  /**
+   * Evaluate the position of the DCM at the beginning of the trajectory.
+   * @param nextDcmIos is the position of the DCM at the beginning of the next step;
+   * @param stepDuration is the duration of the step.
+   */
+  void setDcmIos(const iDynTree::Vector2 &nextDcmIos,
+		 const double &stepDuration);
+    
+ public:
+  /**
+   * Constructor.
+   * @param startTime is the init time of the trajectory;
+   * @param endTime is the end time of the trajectory;
+   * @param zmp is the desired position of the ZMP in the SS phase trajejctory;
+   * @param t0 is the "init" time of the SS phase;
+   * @param omega time constant of the 3D-LIPM.
+   */
+  SingleSupportTrajectory(const double &startTime,
+			  const double &endTime,
+			  const double &t0,
+			  const double &stepDuration,
+			  const double &omega,
+			  const iDynTree::Vector2 &zmp,
+			  const iDynTree::Vector2 &nextDcmIos);
+
+
+  /**
+   * DCM init of step getter.
+   * @return the position of the DCM at the beginning of SS phase.
+   */
+  const iDynTree::Vector2& getDcmIos() const;
+
+  /**
+   * ZMP getter.
+   * @return the position of the ZMP.
+   */
+  const iDynTree::Vector2& getZmp() const;
+  
+  /**
+   * Implementation of the getDcmPos method of the
+   * GeneralSupportTrajectory class.
+   * @param t is the trajectory evaluation time;
+   * @param dcmPos is the cartesian position of the Diverget Component of Motion.
+   * @return true / false in case of success / failure.
+   */
+  bool getDcmPos(const double &t, iDynTree::Vector2& dcmPos) override;
+
+  /**
+   * Implementation of the getDcmVel method of the
+   * GeneralSupportTrajectory class.
+   * @param t is the trajectory evaluation time;
+   * @param dcmVel is cartesian velocity of the Diverget Component of Motion;
+   * @return true / false in case of success / failure.
+   */
+  bool getDcmVel(const double &t, iDynTree::Vector2& dcmVel) override;
+};
+
+/**
+ * This class represents a coordiante (x or y) of the Divergent Component
+ * of Motion trajectory during a double support phase.
+ */
+class DoubleSupportTrajectory : public GeneralSupportTrajectory
+{
+ private:
+  Eigen::Vector4d m_coefficentsX; /**< 3-th order x-trajectory parameters [a3, a2, a1, a0] */
```

Comment:
If you use a fixed-size vectorizable Eigen type (this include `Eigen::Vector4d` as an attribute in a class, you should use the macro `EIGEN_MAKE_ALIGNED_OPERATOR_NEW` to make sure that the new operator works on this class correctly. See https://eigen.tuxfamily.org/dox/group__TopicStructHavingEigenMembers.html for more info.